library/core/src/ptr/mod.rs

```diff
@@ -446,10 +450,11 @@ mod mut_ptr;
 ///   additional invariants - this is type-dependent.
 ///
 /// Additionally, if `T` is not [`Copy`], using the pointed-to value after
-/// calling `drop_in_place` can cause undefined behavior. Note that `*to_drop =
+/// calling `drop_in_place` may cause undefined behavior. Note that `*to_drop =
 /// foo` counts as a use because it will cause the value to be dropped
 /// again. [`write()`] can be used to overwrite data without causing it to be
-/// dropped.
+/// dropped. Read operations may be UB based on library invariants of that type,
+/// for example reading the value pointed to by a dropped `Box<T>` is a use-after-free.
```

Comment:
This sentence seems useful, but can you clarify: Read operations *on what*? Ie, I have code like:

```rust
let p: *mut T = some_pointer_from_somewhere;
drop_in_place(p);
```

I initially interpreted read operations as being about `*p`, but that doesn't seem to agree with the rest of the sentence, which seems to be about `**p` in the case where `T` is `Box<S>`.